src/svd/device.py

```diff
@@ -0,0 +1,1585 @@
+        """
+        :param element: SVD field element binding object.
+        :return: Description of the field.
+        """
+
+        name = element.name
+        bit_range = element.bit_range
+
+        # We do not support "do not care" bits, as by marking bits "x", see
+        # SVD docs "/device/peripherals/peripheral/registers/.../enumeratedValue"
+        if element.enumerated_values is not None:
+            enums = {e.name: e.value for e in element.enumerated_values}
+        else:
+            enums = {}
+
+        allowed_values = set(enums.values()) if enums else range(2**bit_range.width)
+
+        return cls(
+            name=name,
+            bit_range=bit_range,
+            enums=enums,
+            allowed_values=allowed_values,
+            element=element,
+        )
+
+
+FieldParent = TypeVar("FieldParent", Register, FlatRegister)
+
+
+class _Field(Generic[FieldParent]):
+    """
+    Register field instance.
+    Represents a SVD field element.
+    """
+
+    __slots__ = ["_description", "_register", "_allowed_values"]
+
+    def __init__(
+        self,
+        description: _FieldSpec,
+        register: FieldParent,
+    ):
+        """
+        Initialize the class attribute(s).
+
+        :param description: Field description.
+        :param register: Register to which the field belongs.
+        """
+        self._description: _FieldSpec = description
+        self._register: FieldParent = register
+        self._allowed_values: Collection[int] = description.allowed_values
+
+    @property
+    def name(self) -> str:
+        """Name of the field."""
+        return self._description.name
+
+    @property
+    def reset_content(self) -> int:
+        """Default field value."""
+        return self._extract_content_from_register(self._register.reset_content)
+
+    @property
+    def bit_offset(self) -> int:
+        """Bit offset of the field."""
+        return self._description.bit_range.offset
+
+    @property
+    def bit_width(self) -> int:
+        """Bit width of the field."""
+        return self._description.bit_range.width
+
+    @property
+    def mask(self) -> int:
+        """Bitmask of the field."""
+        return ((1 << self.bit_width) - 1) << self.bit_offset
+
+    @property
+    def access(self) -> Access:
+        """Access property of the field."""
+        if (field_access := self._description.element.access) is not None:
+            return field_access
+        return self._register.access
+
+    @property
+    def write_action(self) -> WriteAction:
+        """Side effect of writing to the field."""
+        if (
+            field_write_action := self._description.element.modified_write_values
+        ) is not None:
+            return field_write_action
+        return self._register.write_action
+
+    @property
+    def read_action(self) -> Optional[ReadAction]:
+        """Side effect of reading from the field."""
```

Comment:
nit: Do we have an example of what side effects there would be to reading fields? Reading seems quite harmless in the majority of cases, and I can only think of one instance of half-word-write behavior to some register fields in our products.